components/odh-notebook-controller/controllers/notebook_webhook.go

```diff
@@ -253,8 +265,103 @@ func (w *NotebookWebhook) Handle(ctx context.Context, req admission.Request) adm
 	return admission.PatchResponseFromRaw(req.Object.Raw, marshaledNotebook)
 }
 
+func (w *NotebookWebhook) ClusterWideProxyIsEnabled() bool {
+	proxyResourceList := &configv1.ProxyList{}
+	err := w.Client.List(context.TODO(), proxyResourceList)
+	if err != nil {
+		return false
+	}
+
+	for _, proxy := range proxyResourceList.Items {
+		if proxy.Name == "cluster" {
+			if proxy.Spec.HTTPProxy != "" && proxy.Spec.HTTPSProxy != "" &&
+				proxy.Spec.NoProxy != "" && proxy.Spec.TrustedCA.Name != "" {
+				return true
+			}
+		}
+	}
+	return false
+
+}
+
 // InjectDecoder injects the decoder.
 func (w *NotebookWebhook) InjectDecoder(d *admission.Decoder) error {
 	w.decoder = d
 	return nil
 }
+
+func InjectProxyConfig(notebook *nbv1.Notebook) error {
+	notebookContainers := &notebook.Spec.Template.Spec.Containers
+	var imgContainer corev1.Container
+
+	for _, container := range *notebookContainers {
+		// Update notebook image container
+		if container.Name == notebook.Name {
+			var newVars []corev1.EnvVar
+			imgContainer = container
+
+			for key, val := range proxyEnvVars {
+				keyExists := false
+				for _, env := range imgContainer.Env {
+					if key == env.Name {
+						keyExists = true
+					}
+				}
+				if !keyExists {
+					newVars = append(newVars, corev1.EnvVar{Name: key, Value: val})
+				}
+			}
+
+			// Update container only when required env variables are not present
+			if len(newVars) != 0 {
+				imgContainerExists := false
+				imgContainer.Env = append(imgContainer.Env, newVars...)
+				for index, container := range *notebookContainers {
+					if container.Name == notebook.Name {
+						(*notebookContainers)[index] = imgContainer
+						imgContainerExists = true
+						break
+					}
+				}
+
+				if !imgContainerExists {
+					return fmt.Errorf("notebook image container not found %v", notebook.Name)
+				}
+
+			}
+			break
+		}
+	}
+
+	// Add trusted-ca volume
+	notebookVolumes := &notebook.Spec.Template.Spec.Volumes
```

Comment:
The first question I have is "What is this tls-certificate volume?" 
 "https://github.com/opendatahub-io/kubeflow/blob/24495240d0e896124468509d082e2d57d767c2c8/components/odh-notebook-controller/controllers/notebook_webhook.go#L198-L224

There is no volume mount for this volume in the notebook container.

Reply:
VolumeMount must be added to notebook container and it looks like the following:
~~~
        - name: custom-ca
          volumeMounts:
          - name: trusted-ca
            mountPath: /etc/pki/ca-trust/extracted/pem
            readOnly: true
~~~